meson.build

```diff
@@ -0,0 +1,467 @@
+project('gerbera', 'c', 'cpp',
+  version : '1.9.1',
+  default_options : [ 'c_std=gnu11', 'cpp_std=c++17', 'warning_level=1' ],
+)
+
+cpp = meson.get_compiler('cpp')
+if cpp.get_id() == 'gcc' and cpp.version().version_compare('<8')
+  error('Your GCC version is too old.  You need at least version 8')
+elif cpp.get_id() == 'clang' and cpp.version().version_compare('<7')
+  error('Your clang version is too old.  You need at least version 7.')
+endif
+
+add_global_arguments('-DGERBERA_VERSION="@0@"'.format(meson.project_version()), language : 'cpp')
+add_global_arguments('-DPACKAGE_DATADIR="@0@/@1@/@2@"'.format(get_option('prefix'), get_option('datadir'), meson.project_name()), language : 'cpp')
+
+# This makes assumptions that the libcpp is the GNU one on Linux systems
+if cpp.version().version_compare('>=9')
+  fs_dep = declare_dependency()
+elif host_machine.system() == 'linux'
+  fs_dep = cpp.find_library('stdc++fs')
+elif cpp.get_id() == 'clang'
+  fs_dep = cpp.find_library('c++fs')
+endif
+
+fmt_dep = dependency('fmt', version : '>=6.1.2')
+pugixml_dep = dependency('pugixml')
+spdlog_dep = dependency('spdlog', version : '>=1.5.0')
+sqlite3_dep = dependency('sqlite3', version : '>=3.7.11')
+
+iconv_dep = cpp.find_library('iconv', required : false)
+if not iconv_dep.found()
+  if cpp.has_function('iconv')
```

Comment:
In meson 0.60, you will be able to do `dependency('iconv')` as I've added a custom lookup in https://github.com/mesonbuild/meson/pull/9205 which will first check if iconv_open is available in libc without any dependencies, and then look for `cpp.find_library('iconv')`.

Unfortunately that does require a version so new it is not even released right now. But it will eventually let you simplify these 8 lines down to 1.

Reply:
I'll keep this in mind.